### ISSUE TITLE
Pull DNS servers from public-dns.info nameservers all file

### ISSUE THREAD

**Issue by covuworie** [BBerastegui/fresh-dns-servers#7]
Hi @BBerastegui, me again. Thank you for the update you did last time! 

You say that one of the lists you pull the IP addresses of the DNS servers from is:

- https://public-dns.info/nameservers.txt

I think you should actually pull the list from here instead:

- https://public-dns.info/nameservers-all.txt

Let me explain why. This includes a list of both valid and non-valid DNS servers according to the author of that website. 

Note that the valid status of a DNS server depends on the time when it was last checked (`checked_at` date in the [csv version](https://public-dns.info/nameservers-all.csv) of the file). You will notice that some of the dates are very old so you don't really want to rely on the fact that ONLY the nameservers in https://public-dns.info/nameservers.txt are valid. A great example is the DNS server `1.0.0.1` from Cloudflare which is one of the most reliable and fastest nameservers in the world. It says this DNS server was last checked at `2021-01-10T03:31:57Z`. I'm sure that this DNS server and several others would pass your `dnsvalidator` checks. 

Other reasons for failure (`error` column) in that file are `connection refused` and `timeout`. I've pinged some of these nameservers and they appear to be fine.

You shouldn't have much code to change at all. Just point to the `nameservers-all` file instead of the `nameservers` file. This should lead to a larger list of reliable DNS servers. Let me know if my suggestion sounds reasonable. 

As a side note, I'm not sure why some of the public-dns.info data is so stale when there are only about 15K nameservers to check in total. Your `dnsvalidator` seems to be able to easily check everything and update the results within 24 hours. I may contact the owner of that site for further information.

Thank you.

**Comment by covuworie** [comment]
Looking at [dnsvalidator](https://github.com/frost19k/DNSValidator/blob/master/dnsvalidator/DNSValidator.py) line 178 I see that the `nameservers.txt` file is hardcoded. So this is definitely more than  a single line change in this codebase and as change would be needed there too. 

Also, I had another thought. Are you intending for the checks in `dnsvalidator` to add to  the checks of the `public-dns.info` site or to replace them? If you intend them to add to those checks then I can understand why you chose the `nameservers.txt` file instead of the `nameservers-all.txt` file.

**Comment by BBerastegui** [comment]
Hey @covuworie!

I just committed a change to add the bigger list `https://public-dns.info/nameservers-all.txt` instead. Let's see what we end up with :D

**Comment by BBerastegui** [comment]
Oh, and about the hardcoded `public-dns.info` on the DNSValidator tool, it only pulls that file if you don't provide another set of DNS servers to check, which we are doing in this line here: https://github.com/BBerastegui/fresh-dns-servers/blob/master/.github/workflows/main.yml#L46

Cheers!

**Comment by BBerastegui** [comment]
Well look at this! https://github.com/BBerastegui/fresh-dns-servers/commit/887ec757e4d38c12d5484007e38d0ca68e109795#diff-8e00d75b19285d880c48249a9318d44417349632bad44cebc90abbd4e9b296a1

A sh*t ton of new valid DNS servers as you mentioned @covuworie! Thanks for spotting this, it looks like all the new ones added are also valid :D

Lets keep an eye on this to see how it progresses. Thanks!

**Comment by covuworie** [comment]
@BBerastegui Thanks very much for the prompt changes. Amazing now with all these new valid DNS servers! 

Actually I also added a bunch of new DNS servers that I found yesterday with a web search to [add new nameservers](https://public-dns.info/nameservers/new). About 60 or so were valid out of the thousands I tried. I actually have some more that I will be adding over the next few days so the list should grow a bit larger. If I find anymore in the future I will add them there too. Hopefully with a bit of luck we can get this list to over a thousand DNS servers.

Cheers!
